databricks/koalas/frame.py

```diff
@@ -4008,6 +4008,65 @@ def duplicated(self, subset=None, keep="first") -> "Series":
             )
         )
 
+    def dot(self, other: "Series") -> "Series":
+        """
+        Compute the matrix multiplication between the DataFrame and other.
+
+        This method computes the matrix product between the DataFrame and the
+        values of an other Series
+
+        It can also be called using ``self @ other`` in Python >= 3.5.
+
+        Parameters
+        ----------
+        other : Series
+            The other object to compute the matrix product with.
+
+        Returns
+        -------
+        Series
+            Return the matrix product between self and other as a Series.
+
+        See Also
+        --------
+        Series.dot: Similar method for Series.
+
+        Notes
+        -----
+        The dimensions of DataFrame and other must be compatible in order to
+        compute the matrix multiplication. In addition, the column names of
+        DataFrame and the index of other must contain the same values, as they
+        will be aligned prior to the multiplication.
+
+        The dot method for Series computes the inner product, instead of the
+        matrix product here.
+
+        Examples
+        --------
+        >>> from databricks.koalas.config import set_option, reset_option
+        >>> set_option("compute.ops_on_diff_frames", True)
+        >>> kdf = ks.DataFrame([[0, 1, -2, -1], [1, 1, 1, 1]])
+        >>> kser = ks.Series([1, 1, 2, 1])
+        >>> kdf.dot(kser)
+        0   -4
+        1    5
+        dtype: int64
+
+        Note how shuffling of the objects does not change the result.
+
+        >>> kser2 = kser.reindex([1, 0, 2, 3])
+        >>> kdf.dot(kser2)
+        0   -4
+        1    5
+        dtype: int64
+
+        >>> reset_option("compute.ops_on_diff_frames")
+        """
+        if not isinstance(other, ks.Series):
+            raise TypeError("Unsupported type {}".format(type(other).__name__))
```

Comment:
Seems pandas also support many other types as below, not only `Series`.

```python
>>> pdf
   0  1  2  3
0  0  1 -2 -1
1  1  1  1  1
>>> pser
0    1
1    1
2    2
3    1
dtype: int64
>>>
>>> pdf.dot(pser.to_frame())  # DataFrame
   0
0 -4
1  5
>>> pdf.dot(pser.index)  # Index
0   -6
1    6
dtype: int64
>>> pdf.dot([1, 2, 3, 4])  # list
0    -8
1    10
dtype: int64
```

Reply:
Yeah, I should've clarified in the PR description. This PR targets to support other as `Series` only. 

Considering the implementation complexity and user demand, we might want to support other data types later.

Let me modify the PR description and leave a comment for this.

Thanks for looking into this!

Reply:
Oh, sure I got it :)